core/dbt/contracts/graph/manifest.py

```diff
@@ -437,15 +441,27 @@ def find(
 
         pkg_dct: Mapping[PackageName, List[Any]] = self.storage[search_name]
 
+        nodes = []
         if package is None:
             if not pkg_dct:
                 return None
             else:
-                return next(iter(pkg_dct.values()))
+                nodes = next(iter(pkg_dct.values()))
         elif package in pkg_dct:
-            return pkg_dct[package]
+            nodes = pkg_dct[package]
         else:
             return None
+        if resource_types is None:
+            return nodes
+        else:
+            new_nodes = []
+            for node in nodes:
+                if node.resource_type in resource_types:
+                    new_nodes.append(node)
+            if not new_nodes:
+                return None
```

Comment:
Seems like not covered by any tests, do we need this?

Reply:
This particular line doesn't have functional tests. It would get hit if a resource_types filter is provided and no nodes are returned as a result.

Reply:
Ended up just adding them as part of this PR. Didn't require a lot of state